mitodl/open-discussions
# Import courses and programs via the prolearn search api

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3787 
Related: https://github.com/mitodl/salt-ops/pull/1548

#### What's this PR do?
Adds ETL functions to import courses (and programs if there ever are any) from the prolearn.mit.edu search API, from a certain list of sources (CSAIL, Professional Education, Sloan, Schwarzman College of Computing, Center for Transportation & Logistics).

#### How should this be manually tested?
Set the following in your `.env` file:
```
PROLEARN_CATALOG_API_URL=https://prolearn.mit.edu/graphql
```

Then run:
```
docker-compose run web python manage.py backpopulate_prolearn_data
```

It should complete successfully, after which there should be ~296 new courses, with correct platform, offered_by, image, url values.  Check the `/learn` and `/learn/search` frontend to make sure the new courses display properly and have working urls (there is one course for Schwarzman College of Computing that has a broken url, but that url is coming directly from prolearn).


#### Any background context you want to provide?
There are currently no programs outside of xPro that are included in prolearn results, and those are not processed because we already have a dedicated ETL function for xPro.

#### Screenshots (if appropriate)

#### Offered By filter options:
![Screen Shot 2022-12-12 at 3 37 33 PM](https://user-images.githubusercontent.com/187676/207152527-a607412f-588a-451f-953b-be26a630f4a0.png)

#### Professional Education course:
![Screen Shot 2022-12-12 at 3 37 17 PM](https://user-images.githubusercontent.com/187676/207152615-4a4860bb-ff3c-435f-be8a-d469b25c21a7.png)

#### Sloan School of Management:
![Screen Shot 2022-12-12 at 3 36 56 PM](https://user-images.githubusercontent.com/187676/207152704-4869d3ae-41aa-457a-8875-c227d827693a.png)

#### Bootcamps:
![Screen Shot 2022-12-12 at 3 46 41 PM](https://user-images.githubusercontent.com/187676/207152785-ccfae77f-8d7a-45b9-8c7d-79faff4d33bb.png)

#### Center for Transportation & Logistics
![Screen Shot 2022-12-12 at 3 47 33 PM](https://user-images.githubusercontent.com/187676/207152891-24628f85-76ab-4d89-8cc4-e27620d852d5.png)

#### Schwarzman College
![Screen Shot 2022-12-12 at 3 47 46 PM](https://user-images.githubusercontent.com/187676/207152973-9b0a8c1f-149b-47ed-888c-5ebe002f08e6.png)

#### CSAIL (course has no image)
![Screen Shot 2022-12-12 at 3 48 40 PM](https://user-images.githubusercontent.com/187676/207153005-9ac5cf59-0c99-45db-951b-531095410ad8.png)
